vlc-bin: init at 3.0.20

AndersonTorres opened the pull request:
From https://github.com/NixOS/nixpkgs/pull/209086

## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/3513

AndersonTorres commented:
@ofborg build vlc-bin-arm64 vlc-bin-intel64 vlc-bin-universal

AndersonTorres commented:
@ofborg build vlc-bin-macos-arm64 vlc-bin-macos-intel64 vlc-bin-macos-universal

AndersonTorres commented:
@ofborg build vlc-bin-macos-arm64 vlc-bin-macos-intel64 vlc-bin-macos-universal

SuperSandro2000 commented:
Why are we not building from source? From the linked PR it isn't obvious to me.

AndersonTorres commented:
Because, well, I do not have a Macintosh to play with Darwin idiosyncrasies.

The best I can do for the Nix Darwin users is to repack the binaries provided by upstream.

wegank commented:
I'd actually prefer a single `vlc-bin` attribute to install the intel64 variant on x86_64-darwin and the arm64 variant on aarch64-darwin, instead of `vlc-bin-macos-{arm64,intel64,universal}`.

wegank commented:
> Why are we not building from source? From the linked PR it isn't obvious to me.

The author of that PR mentioned that building VLC requires `ibtool` from Xcode, which has no open-source equivalent.

AndersonTorres commented:
> I'd actually prefer a single `vlc-bin` attribute to install the intel64 variant on x86_64-darwin and the arm64 variant on aarch64-darwin, instead of `vlc-bin-macos-{arm64,intel64,universal}`.

Done. What do you think?

pcasaretto commented:
Result of `nixpkgs-review pr 291405` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vlc-bin-macos (vlc-bin-macos-universal)</li>
  </ul>
</details>

pcasaretto commented:
Tested VLC app and working as expected.

AndersonTorres commented:
@ofborg build vlc-bin

wegank commented on pkgs/by-name/vl/vlc-bin-macos/package.nix:
```diff
@@ -0,0 +1,55 @@
+    hash = {
+      "arm64" = "sha256-XV8O5S2BmCpiL0AhkopktHBalVRJniDDPQusIlkLEY4=";
+      "intel64" = "sha256-pNwUQfyrjiuQxil0zlOZu4isv2xw1U8hxhWNn7H7onk=";
+      "universal" = "sha256-IqGPOWzMmHbGDV+0QxFslv19BC2J1Z5Qzcuja/Od1Us=";
```
Is it really necessary to provide a universal one, if there are native versions for both platforms?

AndersonTorres replied:
I don't know. I included it for completeness.

wegank commented on pkgs/by-name/vl/vlc-bin-macos/package.nix:
```diff
@@ -0,0 +1,55 @@
+    homepage = "https://www.videolan.org/vlc/";
+    downloadPage = "https://www.videolan.org/vlc/download-macosx.html";
+    license = lib.licenses.lgpl21Plus;
+    maintainers = with lib.maintainers; [ ];
```
Probably add someone here?

AndersonTorres replied:
Since I am not a Darwin user, and  I don't know anyone interested, I am putting this package to adoption.

SuperSandro2000 replied:
If we don't find someone for this, we shouldn't merge this PR.

pcasaretto replied:
I'd gladly maintain it. 😃

AndersonTorres replied:
I will add you to maintainers set, then!

wegank commented on pkgs/top-level/all-packages.nix:
```diff
@@ -35972,6 +35972,15 @@ with pkgs;
   vkeybd = callPackage ../applications/audio/vkeybd { };
 
+  vlc-bin = vlc-bin-macos.override {
+    variant = if (stdenv.isDarwin && stdenv.isAarch64)
+              then "arm64"
+              else if (stdenv.isDarwin && stdenv.isx86_64)
+              then "intel64"
+              else (throw "Unsupported system: ${stdenv.hostPlatform.system}");
```
I'd move that logic to `pkgs/by-name/vl/vlc-bin-macos/package.nix` (but is the `-macos` suffix really necessary? The pattern seems unprecedented in Nixpkgs.)

AndersonTorres replied:
I wanted to make it clear that it is a Macos-only binary.

AndersonTorres committed:
vlc-bin: init at 3.0.20

Taken from https://github.com/NixOS/nixpkgs/pull/209086

Since I (AndersonTorres) am not a Darwin user, I will not add myself to
maintainers. This package is ready and open for adoptions!

Co-authored-by: willemml <willem@leit.so>

AndersonTorres committed:
maintainers: add pcasaretto

Because he wants to keep care of vlc-bin package!

AndersonTorres committed:
vlc-bin: add pcasaretto to meta.maintainers

As seen in https://github.com/NixOS/nixpkgs/pull/291405#discussion_r1507465727

AndersonTorres commented:
I think it is now OK!

wegank commented:
Result of `nixpkgs-review pr 291405` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>vlc-bin</li>
    <li>vlc-bin-universal</li>
  </ul>
</details>

wegank commented:
Result of `nixpkgs-review pr 291405` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>vlc-bin</li>
    <li>vlc-bin-universal</li>
  </ul>
</details>

wegank commented:
@ofborg build vlc-bin vlc-bin.passthru.tests

SuperSandro2000 commented on pkgs/by-name/vl/vlc-bin/package.nix:
```diff
@@ -0,0 +1,61 @@
+    license = lib.licenses.lgpl21Plus;
+    mainProgram = "vlc";
+    maintainers = with lib.maintainers; [ pcasaretto ];
+    platforms = lib.systems.inspect.patternLogicalAnd
+      (lib.systems.inspect.patterns.isDarwin)
+      (({
+        "arm64" = lib.systems.inspect.patterns.isAarch64;
+        "intel64" = lib.systems.inspect.patterns.isx86_64;
+        "universal" = lib.systems.inspect.patterns.isDarwin;
+      }).${variant});
```
Please just list the attrs as a list of strings, There should be only really two here anyway.

pcasaretto commented on maintainers/maintainer-list.nix:
```diff
@@ -14852,6 +14852,12 @@
+    email = "pcasaretto@gmail.com";
+    github = "pcasaretto";
+    githubId = 817039;
+    name = "Paulo L F Casaretto";
```
```suggestion
    name = "Paulo Casaretto";
```
A small request.

AndersonTorres replied:
As you wish!
